compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -734,3 +782,42 @@ internal val KSDeclarationContainer.declarationsInSourceOrder: Sequence<KSDeclar
         return (declarations as? Sequence<KSDeclarationDescriptorImpl>)?.sortedWith(declarationOrdering.comparator)
             ?: declarations
     }
+
+internal val KSPropertyDeclaration.jvmAccessFlag: Int
+    get() = when (origin) {
+        Origin.KOTLIN_LIB -> {
+            val descriptor = (this as KSPropertyDeclarationDescriptorImpl).descriptor
+            val kotlinBinaryJavaClass = descriptor.getContainingKotlinJvmBinaryClass()
+            kotlinBinaryJavaClass?.let {
+                BinaryClassInfoCache.getCached(it).fieldAccFlags.get(this.simpleName.asString()) ?: 0
+            } ?: 0
```

Comment:
not sure if return 0 is best here, I feel if the containing class can't be obtained, it rather implies an error?

Reply:
`getContainingKotlinJvmBinaryClass()` has null checks everywhere and I'm not so confident with it. For example, I got a null for some `<init>` of `Companion`. Also there can be some builtin functions, e.g., `Boolean.equals` that doesn't have sources.

Reply:
Added a few comments in code.